quarto/catalog/fda-table_03/index.qmd

```diff
@@ -0,0 +1,195 @@
+```
+
+## Build ARD
+
+```{r ard, message=FALSE, warning=FALSE, results='hide'}
+make_table_03 <- function(df,
+                          denominator = NULL,
+                          return_ard = TRUE,
+                          arm_var = "ARM",
+                          id_var = "USUBJID",
+                          scrnfl_var = "SCRNFL",
+                          scrnfailfl_var = "SCRNFAILFL",
+                          scrnfail_var = "SCRNFRS") {
+  ard <- ard_table_03(
+    df = df,
+    denominator = denominator,
+    arm_var = arm_var,
+    id_var = id_var,
+    scrnfl_var = scrnfl_var,
+    scrnfailfl_var = scrnfailfl_var,
+    scrnfail_var = scrnfail_var
+  )
+
+  tbl <- make_table_03_gtsummary(
+    df = df,
+    denominator = denominator,
+    arm_var = arm_var,
+    id_var = id_var,
+    scrnfl_var = scrnfl_var,
+    scrnfailfl_var = scrnfailfl_var,
+    scrnfail_var = scrnfail_var
+  )
+
+  if (return_ard) {
+    return(list(table = tbl, ard = ard))
+  } else {
+    return(tbl)
+  }
+}
+
+preproc_df_table_03 <- function(df,
+                                arm_var = "ARM",
+                                id_var = "USUBJID",
+                                scrnfl_var = "SCRNFL",
+                                scrnfailfl_var = "SCRNFAILFL",
+                                scrnfail_var = "SCRNFRS") {
+  df %>%
+    df_explicit_na() %>%
+    mutate(
+      SCRNFL = with_label(.data[[scrnfl_var]] == "Y", "Patients screened"),
+      SCRNFL = with_label(!is.na(scrnfail_var), "Screnning failures"),
+      ENRLFL = with_label(!is.na(ENRLDT), "Patients enrolled"),
+      RANDFL = with_label(!is.na(RANDDT), "Patients randomized")
+    )
+}
+
+ard_table_03 <- function(df,
+                         denominator = NULL,
+                         arm_var = "ARM",
+                         id_var = "USUBJID",
+                         scrnfl_var = "SCRNFL",
+                         scrnfailfl_var = "SCRNFAILFL",
+                         scrnfail_var = "SCRNFRS") {
+  df <- preproc_df_table_03(df, arm_var, id_var, scrnfl_var, scrnfailfl_var, scrnfail_var)
+
+  if (is.null(denominator)) {
+    denominator <- df
+  } else {
+    denominator <- alt_counts_df_preproc(denominator, id_var, arm_var)
+  }
+
+  scrn_ard <- df |>
+    group_by(!!rlang::sym(arm_var)) |>
+    ard_dichotomous(
+      variables = SCRNFL,
+      value = list(SCRNFL = TRUE),
+      statistic = everything() ~ c("n"),
+      denominator = denominator
+    )
+
+  scrnfail_ard <- df |>
+    group_by(!!rlang::sym(arm_var)) |>
+    ard_categorical(
+      variables = SCRNFRS,
+      statistic = everything() ~ c("n", "p")
+    )
+
+  enrl_rand_ard <- df |>
+    group_by(!!rlang::sym(arm_var)) |>
+    ard_dichotomous(
+      variables = c(ENRLFL, RANDFL),
+      value = list(ENRLFL = TRUE, RANDFL = TRUE),
+      statistic = everything() ~ c("n", "p"),
+      denominator = denominator
+    )
+
+  ard <- bind_ard(scrn_ard, scrnfail_ard, enrl_rand_ard)
```

Comment:
I think using ard_stack() instead would allow you to make all 3 ARDs without defining the arm_var every time